core/src/main/java/org/apache/accumulo/core/file/blockfile/cache/TinyLfuBlockCache.java

```diff
@@ -48,12 +52,13 @@
   private final Cache<String,Block> cache;
   private final Policy.Eviction<String,Block> policy;
   private final ScheduledExecutorService statsExecutor;
+  private final Map<String,CacheEntry> weakEvictionMap = new MapMaker().weakValues().makeMap();
 
   public TinyLfuBlockCache(long maxSize, long blockSize) {
     cache = Caffeine.newBuilder().initialCapacity((int) Math.ceil(1.2 * maxSize / blockSize)).weigher((String blockName, Block block) -> {
       int keyWeight = ClassSize.align(blockName.length()) + ClassSize.STRING;
       return keyWeight + block.weight();
-    }).maximumWeight(maxSize).recordStats().build();
+    }).maximumWeight(maxSize).recordStats().removalListener((String key, Block block, RemovalCause cause) -> weakEvictionMap.put(key, block)).build();
```

Comment:
Do you know if the removal listener is called before or after the entry is removed from the cache?